ethcore/src/executed.rs

```diff
@@ -23,172 +23,181 @@ use vm;
+
+    /// Gas refunded after the execution of transaction.
+    /// To get gas that was required up front, add `refunded` and `gas_used`.
+    pub refunded: U256,
+
+    /// Cumulative gas used in current block so far.
+    ///
+    /// `cumulative_gas_used = gas_used(t0) + gas_used(t1) + ... gas_used(tn)`
+    ///
+    /// where `tn` is current transaction.
+    pub cumulative_gas_used: U256,
+
+    /// Vector of logs generated by transaction.
+    pub logs: Vec<LogEntry>,
+
+    /// Addresses of contracts created during execution of transaction.
+    /// Ordered from earliest creation.
+    ///
+    /// eg. sender creates contract A and A in constructor creates contract B
+    ///
+    /// B creation ends first, and it will be the first element of the vector.
+    pub contracts_created: Vec<Address>,
+    /// Transaction output.
+    pub output: Bytes,
+    /// The trace of this transaction.
+    pub trace: Vec<T>,
+    /// The VM trace of this transaction.
+    pub vm_trace: Option<V>,
+    /// The state diff, if we traced it.
+    pub state_diff: Option<StateDiff>,
+}
+
+impl HeapSizeOf for Executed {
+    fn heap_size_of_children(&self) -> usize {
+        self.logs.heap_size_of_children()
+            + self.contracts_created.heap_size_of_children()
+            + self.trace.heap_size_of_children()
+    }
 }
 
 /// Result of executing the transaction.
 #[derive(PartialEq, Debug, Clone)]
 pub enum ExecutionError {
-	/// Returned when there gas paid for transaction execution is
-	/// lower than base gas required.
-	NotEnoughBaseGas {
-		/// Absolute minimum gas required.
-		required: U256,
-		/// Gas provided.
-		got: U256
-	},
-	/// Returned when block (gas_used + gas) > gas_limit.
-	///
-	/// If gas =< gas_limit, upstream may try to execute the transaction
-	/// in next block.
-	BlockGasLimitReached {
-		/// Gas limit of block for transaction.
-		gas_limit: U256,
-		/// Gas used in block prior to transaction.
-		gas_used: U256,
-		/// Amount of gas in block.
-		gas: U256
-	},
-	/// Returned when transaction nonce does not match state nonce.
-	InvalidNonce {
-		/// Nonce expected.
-		expected: U256,
-		/// Nonce found.
-		got: U256
-	},
-	/// Returned when cost of transaction (value + gas_price * gas) exceeds
-	/// current sender balance.
-	NotEnoughCash {
-		/// Minimum required balance.
-		required: U512,
-		/// Actual balance.
-		got: U512
-	},
-	/// When execution tries to modify the state in static context
-	MutableCallInStaticContext,
-	/// Returned when transacting from a non-existing account with dust protection enabled.
-	SenderMustExist,
-	/// Returned when internal evm error occurs.
-	Internal(String),
-	/// Returned when generic transaction occurs
-	TransactionMalformed(String),
+    /// Returned when there gas paid for transaction execution is
+    /// lower than base gas required.
+    NotEnoughBaseGas {
+        /// Absolute minimum gas required.
+        required: U256,
+        /// Gas provided.
+        got: U256
```

Comment:
we use tabs instead of spaces. Please make sure that you code editor utilizes `.editorconfig` and revert these unnecessary changes